nullaway/src/test/java/com/uber/nullaway/NullAwayContractsTests.java

```diff
@@ -267,4 +267,161 @@ public void customContractAnnotation() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void contractDeclaringBothNotNull() {
+    makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber"))
+        .addSourceLines(
+            "NullnessChecker.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import org.jetbrains.annotations.Contract;",
+            "public class NullnessChecker {",
+            "  @Contract(\"null, _ -> false; _, null -> false\")",
+            "  static boolean bothNotNull(@Nullable Object o1, @Nullable Object o2) {",
+            "    // null, _ -> false",
+            "    if (o1 == null) {",
+            "      return false;",
+            "    }",
+            "    // _, null -> false",
+            "    if (o2 == null) {",
+            "      return false;",
+            "    }",
+            "    // Function cannot declare a contract for true",
+            "    return System.currentTimeMillis() % 100 == 0;",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "class Test {",
+            "  String test1(@Nullable Object o) {",
```

Comment:
Do we want two test methods here? One to check the first argument to `bothNotNull` and another the second?